File: katas/6 kyu/does-my-number-look-big-in-this/solution.py
# does-my-number-look-big-in-this
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def narcissistic( value ):
  sum = 0
  pow = len(str(value))
  for digit in str(value):
    sum += int(digit) ** pow
  return True if sum == value else False
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def narcissistic(value):
  return value == sum(int(x) ** len(str(value)) for x in str(value))
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(narcissistic(7), True, '7 is narcissistic');
test.assert_equals(narcissistic(371), True, '371 is narcissistic');
test.assert_equals(narcissistic(122), False, '122 is not narcissistic')
test.assert_equals(narcissistic(4887), False, '4887 is not narcissistic')

File: katas/6 kyu/does-my-number-look-big-in-this/solution.php
<!-- does-my-number-look-big-in-this -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function narcissistic(int $value): bool {
  $string = strval($value);
  $pow = strlen($string);
  $result = array();
  foreach (str_split($string) as $digit) {
    array_push($result , $digit**$pow);
  }
  return array_sum($result) == $value ? true : false;
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function narcissistic(int $value): bool {
  $digitCount = strlen($value);
  
  foreach(str_split($value) as $digit){
    $value -= pow($digit, $digitCount);
  }
  
  return $value === 0;
}
/* ---------------------------------- TEST ---------------------------------- */
class NarcissisticTest extends TestCase {
  public function testExamples() {
    $this->assertTrue(narcissistic(7), '7 is narcissistic');
    $this->assertTrue(narcissistic(371), '371 is narcissistic');
  }
}